imap_data_access/file_naming.py

```diff
@@ -0,0 +1,33 @@
+"""Methods for managing and validating filenames."""
+import re
+
+def extract_filename_components(filename: str):
+    """
+    Extracts all components from filename.
+
+    Parameters
+    ----------
+    filename : str
+        Path of dependency data.
```

Comment:
Do we want to accept paths here too? Might be able to handle them easily enough with: `filename = Path(filename).name`

```suggestion
    filename : pathlib.Path or str
        Name of the file to extract components from
```